src/main/java/thrift/MainApp.java

```diff
@@ -160,8 +161,22 @@ protected void initCurrencyMappings(Storage storage) {
             logger.warning("Currency Mappings file at " + currencyMappingsFilePath
                     + " is not in the correct format. "
                     + "Using default currency mappings");
+
+            Map<String, Double> defaultCurrencyMappings = CurrencyUtil.getCurrencyMap();
+            try {
+                storage.saveCurrencyMappings(defaultCurrencyMappings);
+            } catch (IOException ioe) {
+                //do nothing
```

Comment:
Should throw an exception and possibly a `logger.error()`

Reply:
Also, is there a more elegant way than having a nested try-catch block? 🤔

Reply:
The logic has been changed to save the file regardless of whether a valid file exists or not.